pkg/plugins/policies/meshretry/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,166 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	common_api "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
+)
+
+func (r *MeshRetryResource) validate() error {
+	var verr validators.ValidationError
+	path := validators.RootedAt("spec")
+	verr.AddErrorAt(path.Field("targetRef"), validateTop(r.Spec.TargetRef))
+	if len(r.Spec.To) == 0 {
+		verr.AddViolationAt(path.Field("to"), "needs at least one item")
+	}
+	verr.AddErrorAt(path, validateTo(r.Spec.To))
+	return verr.OrNil()
+}
+
+func validateTop(targetRef common_api.TargetRef) validators.ValidationError {
+	targetRefErr := matcher_validators.ValidateTargetRef(targetRef, &matcher_validators.ValidateTargetRefOpts{
+		SupportedKinds: []common_api.TargetRefKind{
+			common_api.Mesh,
+			common_api.MeshSubset,
+			common_api.MeshService,
+			common_api.MeshServiceSubset,
+		},
+	})
+	return targetRefErr
+}
+
+func validateTo(to []To) validators.ValidationError {
+	var verr validators.ValidationError
+	for idx, toItem := range to {
+		path := validators.RootedAt("to").Index(idx)
+		verr.AddErrorAt(path.Field("targetRef"), matcher_validators.ValidateTargetRef(toItem.TargetRef, &matcher_validators.ValidateTargetRefOpts{
+			SupportedKinds: []common_api.TargetRefKind{
+				common_api.Mesh,
+				common_api.MeshService,
+			},
+		}))
+		verr.AddErrorAt(path.Field("default"), validateDefault(toItem.Default))
+	}
+	return verr
+}
+
+func validateDefault(conf Conf) validators.ValidationError {
+	var verr validators.ValidationError
+	path := validators.RootedAt("conf")
+	if conf.TCP == nil && conf.HTTP == nil && conf.GRPC == nil {
```

Comment:
shouldn't it be legal to set empty `tcp: {}` or `http: {}` to disable retry?
We introduced `disabled: true` in MeshCircuitBreaker. I wonder if we should do the same here.

Reply:
you can disable retries with:
```
http:
  numRetries: 0
grpc:
  numRetries: 0
```

Reply:
nice. I think it would be useful to include this in docs